src/components/Comments/WriteComment.js

```diff
@@ -58,8 +60,21 @@ const WriteComment = ({
   const toggleSwitch = () => {
     setIsAnonimity((prevState) => !prevState);
     getAnonUser();
+    if (isAnonimity) saveToStorage('null');
+    if (!isAnonimity) saveToStorage('true');
   };
 
+  React.useEffect(() => {
+    AsyncStorage.getItem(storageKey).then((data) => {
+      if (data === 'true') {
+        setIsAnonimity(true);
+        getAnonUser();
+      }
+    });
+  }, []);
+  const saveToStorage = (valueData) => {
+    AsyncStorage.setItem(storageKey, valueData);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block introduces a new feature to save the anonymity state of the user in local storage. However, there's a potential issue with the way it's implemented. The `saveToStorage` function is defined after it's used in `toggleSwitch`, which could lead to a ReferenceError if `toggleSwitch` is called before `saveToStorage` is defined. To fix this, you should move the definition of `saveToStorage` above its usage in `toggleSwitch`.

```diff
-  const toggleSwitch = () => {
-    setIsAnonimity((prevState) => !prevState);
-    getAnonUser();
-    if (isAnonimity) saveToStorage('null');
-    if (!isAnonimity) saveToStorage('true');
-   };
-  
-  React.useEffect(() => {
-    AsyncStorage.getItem(storageKey).then((data) => {
-      if (data === 'true') {
-        setIsAnonimity(true);
-        getAnonUser();
-      }
-    });
-  }, []);
-  const saveToStorage = (valueData) => {
-    AsyncStorage.setItem(storageKey, valueData);
-  };

+  const saveToStorage = (valueData) => {
+    AsyncStorage.setItem(storageKey, valueData);
+  };
+
+  const toggleSwitch = () => {
+    setIsAnonimity((prevState) => !prevState);
+    getAnonUser();
+    if (isAnonimity) saveToStorage('null');
+    if (!isAnonimity) saveToStorage('true');
+   };
+  
+  React.useEffect(() => {
+    AsyncStorage.getItem(storageKey).then((data) => {
+      if (data === 'true') {
+        setIsAnonimity(true);
+        getAnonUser();
+      }
+    });
+  }, []);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->